ui/src/services/k8s/solutions.js

```diff
@@ -96,18 +98,57 @@ export async function getEnvironmentAdminUIs(environment) {
   }));
 }
 
-export async function getEnvironmentConfigMap(environment) {
-  for (const namespace of environment.namespaces) {
-    const result = await coreV1.readNamespacedConfigMap(
+export async function getNamespacedConfigmap(environment) {
+  try {
+    return await coreV1.readNamespacedConfigMap(
       ENVIRONMENT_CONFIGMAP_NAME,
-      namespace,
+      environment,
     );
+  } catch (error) {
+    return { error };
+  }
+}
 
-    if (!result.error) {
-      // Return the first one found among the Environment namespaces
-      return { ...result?.data };
-    }
+export async function createNamespacedConfigMap(namespace) {
+  const body = {
+    metadata: {
+      name: ENVIRONMENT_CONFIGMAP_NAME,
+    },
+  };
+  try {
+    return await coreV1.createNamespacedConfigMap(namespace, body);
+  } catch (error) {
+    return { error };
   }
 }
 
+export async function patchNamespacedConfigMap(namespace, data) {
+  const body = {
+    data: { ...data },
+  };
+  try {
+    return await coreV1.patchNamespacedConfigMap(
+      ENVIRONMENT_CONFIGMAP_NAME,
+      namespace,
+      body,
+      undefined,
+      undefined,
+      {
+        headers: {
+          'Content-Type': 'application/merge-patch+json',
+        },
+      },
+    );
+  } catch (error) {
+    return { error };
+  }
+}
```

Comment:
Same principles here: define the generic "patch ConfigMap" method in `core`, and build on it in `solutions` for just your needs!

```javascript
export async function addSolutionToEnv(namespace, solName, solVersion) {
  const patch = { op: "add", path: `/data/${solName}`, value: solVersion };

  return core.patchConfigMap(
    ENVIRONMENT_CONFIGMAP_NAME, namespace,
    // use an object instead of many arguments, for easier support of
    // multiple methods (e.g. both `jsonPatch` and `mergePatch` could be
    // allowed, headers being inferred from which is used).
    { jsonPatch = [patch] },
  );
}

export async function updateSolutionInEnv(namespace, solName, solVersion) {
  const patch = { op: "replace", path: `/data/${solName}`, value: solVersion };

  return core.patchConfigMap(
    ENVIRONMENT_CONFIGMAP_NAME, namespace, { jsonPatch = [patch] },
  );
}

export async function removeSolutionFromEnv(namespace, solName) {
  const patch = { op: "remove", path: `/data/${solName}` };

  return core.patchConfigMap(
    ENVIRONMENT_CONFIGMAP_NAME, namespace, { jsonPatch = [patch] },
  );
}
```

Reply:
Good to know the jsonPatch!!!